src/rpc/mod.rs

```diff
@@ -0,0 +1,153 @@
+pub mod eth;
+
+mod helpers {
+    use crate::{
+        accessors::chain,
+        kv::{
+            mdbx::{EnvironmentKind, MdbxTransaction, TransactionKind},
+            tables,
+        },
+        models::{BlockNumber, Message, TransactionAction},
+        stagedsync::stages,
+    };
+    use ethereum_jsonrpc::types;
+    use ethereum_types::U64;
+
+    fn block_number_to_u64(block_number: types::BlockNumber) -> u64 {
+        match block_number {
+            types::BlockNumber::Number(n) => n.as_u64(),
+            _ => 0,
+        }
+    }
+    pub fn get_block_number<K: TransactionKind, E: EnvironmentKind>(
+        txn: &MdbxTransaction<'_, K, E>,
+        block_number: ethereum_jsonrpc::types::BlockId,
+    ) -> anyhow::Result<BlockNumber> {
+        let block_number = match block_number {
+            types::BlockId::Number(n) => BlockNumber(block_number_to_u64(n)),
+            _ => txn
+                .get(tables::SyncStage, stages::FINISH)?
+                .unwrap_or(BlockNumber(0)),
+        };
+        Ok(block_number)
+    }
+
+    pub fn construct_block<K: TransactionKind, E: EnvironmentKind>(
+        txn: &MdbxTransaction<'_, K, E>,
+        block_id: types::BlockId,
+        include_txs: bool,
+        uncle_index: Option<U64>,
+    ) -> anyhow::Result<types::Block> {
+        let (block_number, block_hash, header) = match block_id {
+            types::BlockId::Number(n) => {
+                let block_number =
+                    get_block_number(txn, ethereum_jsonrpc::types::BlockId::Number(n))?;
+                let block_hash = txn.get(tables::CanonicalHeader, block_number)?.unwrap();
+
+                match uncle_index {
+                    Some(n) => {
+                        let body = txn
+                            .get(tables::BlockBody, (block_number, block_hash))?
+                            .unwrap();
```

Comment:
Please check unwrap usage. Invalid RPC parameters should be handled properly, not bump into unwraps.